### ISSUE TITLE
Build fails on Opensuse Leap 15.2

### ISSUE THREAD

**Issue by Charadon** [dankamongmen/notcurses#806]
I can't build notcurses on opensuse, it fails with errors that seem to be related to ncurses.

The cmake command I used was
`cmake -DUSE_MULTIMEDIA=ffmpeg -DDFSG_BUILD=on ..`

Here's the output of make:
[output.txt](https://github.com/dankamongmen/notcurses/files/4944609/output.txt)

I tried building with Ninja, and with OpenImage instead of ffmpeg also, but it gave the same errors.

**Details:**
- GCC Version: 7.5.0 ( but it fails on 8, 9, and 10 )
- Kernel: 5.3.18-lp152.20.7-default
- Cmake Version: 3.17.0
- Make Version: 4.2.1
- Ncurses version: 6.1
- FFMPEG version: 4.2.1
- OpenImage Version: 2.1.17

Just ask if you want me to provide more details.

Edit: Forgot to mention, it also fails to build on OpenSUSE Tumbleweed.

**Comment by dankamongmen** [comment]
Thanks for the report! I'll take a look into this ASAP.

**Comment by dankamongmen** [comment]
Here's the error from the output you posted:

```
In file included from /usr/include/ncurses/term.h:47:0,
                 from /mnt/conglomerate/productive/packages/notcurses/notcurses-1.6.3/src/lib/internal.h:23,
                 from /mnt/conglomerate/productive/packages/notcurses/notcurses-1.6.3/src/lib/notcurses.c:3:
/mnt/conglomerate/productive/packages/notcurses/notcurses-1.6.3/src/lib/notcurses.c: In function ‘create_initial_ncplane’:
/usr/include/ncurses/ncurses.h:1456:21: error: ‘notcurses {aka struct notcurses}’ has no member named ‘_nc_stdscr’; did you mean ‘stdscr’?
```
So what this comes down to is how ncurses is exporting `stdscr`. I'm guessing that it's some kind of macro...

```
 #define stdscr      NCURSES_PUBLIC_VAR(stdscr())
                     ^
```

and indeed it is. Alright. Hrmmm, that's the same way ncurses.h exports it on my Debian machine...

```
/usr/include/ncurses.h:NCURSES_WRAPPED_VAR(WINDOW *, stdscr);
/usr/include/ncurses.h:#define stdscr      NCURSES_PUBLIC_VAR(stdscr())
/usr/include/ncurses.h:extern NCURSES_EXPORT_VAR(WINDOW *) stdscr;
```

hrmmm, let me dig into this a bit deeper. this arises from `internal.h` including `term.h`, which in turn includes `ncurses.h`, apparently.

**Comment by dankamongmen** [comment]
ok, it looks like the `term.h` on Debian doesn't include `ncurses.h`, and thus we don't run into this problem.

**Comment by dankamongmen** [comment]
Bah, I think the easiest way to fix this will be just renaming stdscr to stdplane. I don't have an OpenSUSE installation to test anything more advanced on. @Charadon , if I make a change to head, can you test it?

**Comment by dankamongmen** [comment]
I've made the proposed change. If you could let me know whether it now builds, or at least if you get a new error, I'd appreciate it!

**Comment by Charadon** [comment]
Sure, but you can also use the opensuse's OBS (Open Build Service) if you're wanting to see if it compiles. Unless you mean running the demo programs and seeing if stuff functions properly.

**Comment by Charadon** [comment]
Okay, I get a new error, but atleast it's not the stdscr one.
[output.txt](https://github.com/dankamongmen/notcurses/files/4945544/output.txt)
Now it's complaining about not allowing dynamic exception specifications.

**Comment by Charadon** [comment]
My last comment was with oiio. This output was with ffmpeg, and it gives an error:
[output.txt](https://github.com/dankamongmen/notcurses/files/4945561/output.txt)
`error: initializer element is not constant static int writeline = HUD_ROWS - 2;`

**Comment by dankamongmen** [comment]
on it, thanks

**Comment by dankamongmen** [comment]
I believe the `HUD_ROWS` issue ought be resolved. Looking into the other. I wonder why OpenSUSE is so much weirder than all the other distros I've ported to :/.

**Comment by dankamongmen** [comment]
> Okay, I get a new error, but atleast it's not the stdscr one.
> [output.txt](https://github.com/dankamongmen/notcurses/files/4945544/output.txt)
> Now it's complaining about not allowing dynamic exception specifications.

This is an issue with OpenImageIO, so far as I can tell:

```
/usr/include/OpenEXR/ImathVec.h:771:37: error: ISO C++1z does not allow dynamic
 Vec3<short>::normalizedExc () const throw (IEX_NAMESPACE::MathExc);
                                     ^~~~~
```

maybe I can add some flag to work around this, but I doubt it. FFMpeg is the recommended engine, so let's see if we can't make that work. I appreciate you bearing with me here!

**Comment by Charadon** [comment]
Good news. it compiles now with ffmpeg. I'm gonna run it through the OBS and see if it has any objections to it. Only thing I had to add was -DBUILD_TESTING=OFF because no one in the entire opensuse sphere has packaged doctest. (Maybe I should do that lol)

**Comment by dankamongmen** [comment]
w000t! I don't think I can do anything about the OIIO case, alas. So long as ffmpeg is working, that's enough for me.

I recommend running `notcurses-demo -p ../data` to check the result. For best results, make sure you:

* have a correct `TERM` for your terminal emulator
* are using the 256-color or direct- variant of `TERM`, if one is applicable
* defining `COLORTERM=24bit` if necessary and applicable
* that you have a valid UTF8 locale specified by your `LANG`

check for more info: https://github.com/dankamongmen/notcurses/blob/master/TERMS.md

Go ahead and close this bug if you're satisfied. Thanks!

**Comment by Charadon** [comment]
Okay, so after a normal tussle with cmake (I hate cmake) I finally got it packaged on the OBS
https://build.opensuse.org/package/show/home:Charadon/notcurses
It doesn't contain a couple of things such as static stuff.

I ran `notcurses-demo` and all tests semed to have passed. As a side not, i'd recommend you add a seizure warning to the demos, because some of them were too much even for me who doesn't have epilepsy, and I had to look away multiple times lol

**notcurses-demo report:**
[output.txt](https://github.com/dankamongmen/notcurses/files/4946182/output.txt)


I'm very interested to see where this project goes, because those demos were impressive.
